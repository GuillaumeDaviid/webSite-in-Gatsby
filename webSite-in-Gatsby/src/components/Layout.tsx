import * as React from 'react'
import { Link, useStaticQuery, graphql } from 'gatsby'
// @ts-ignore
import favicon from '../images/favicon.ico'
import  {Helmet} from "react-helmet"
import './styles/Layout.scss'

const Layout = ({ pageTitle, children }:any) => {

    return(
        <div className='Layout'>
          <Helmet>
    <html lang="fr" />
    <link rel="icon" href={favicon}/>
    <title>Guillaume David | {pageTitle}</title>
    </Helmet>
          <div className='Layout__nav'>
            <nav>
        <ul className='Layout__nav--children' >
          <li className='nav__elt'><Link to="/" >Home</Link></li>
          <li className='nav__elt'><Link to="/about">About</Link></li>
          <li className='nav__elt'><Link to="/blog">Blog</Link></li>
        </ul>
      </nav>
      <h1 className='Layout__title'>{pageTitle}</h1>
      <div className='Header_separator-white'></div>
      </div>
      <main>
        {children}
      </main>
        </div>
    )
}

export default Layout