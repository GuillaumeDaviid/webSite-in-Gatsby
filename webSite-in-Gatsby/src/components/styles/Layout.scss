.Layout {
    &__nav {
        background: url('../../images/homebg.jpg') no-repeat 100% 100%;
        background-size: cover;
        position: relative;
        height: 750px;
        color: white;
        &--children {
            margin-right: 100px;
            display: flex;
            justify-content: center;
            align-items: center;
        }
    }

    &__title {
        margin-top: 260px;
        display: flex;
        font-size: 3rem;
        justify-content: center;
        align-items: center;
    }
}

.nav__elt {
    font-size: 1rem;
    margin-left: 100px;
    margin-top: 10px;
}