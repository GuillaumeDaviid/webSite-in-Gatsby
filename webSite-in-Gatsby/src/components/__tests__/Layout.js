import React from "react"
import renderer from "react-test-renderer"
import Layout from "../Layout.tsx"
import { render, screen } from "@testing-library/react"

describe("Layout", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Layout />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should have a title element", () => {
    render(<Layout />)

    const title = screen.getByTestId('title');
    expect(title).toBeTruthy();
  })

  it("should have a title element", () => {
    render(<Layout />)

    const separator = screen.getByTestId('separator');
    expect(separator).toBeTruthy();
  })

  it("should have a title element", () => {
    render(<Layout />)

    const nav = screen.getByTestId('nav');
    expect(nav).toBeTruthy();
  })

  it("should have a title element", () => {
    render(<Layout />)

    const li = screen.getAllByTestId('li');
    expect(li).toHaveLength(3);
  })
})