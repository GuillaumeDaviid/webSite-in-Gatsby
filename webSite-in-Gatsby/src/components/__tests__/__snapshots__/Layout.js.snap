// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  className="Layout"
>
  <div
    className="Layout__nav"
    data-testid="nav"
  >
    <nav>
      <ul
        className="Layout__nav--children"
      >
        <li
          className="nav__elt"
          data-testid="li"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          className="nav__elt"
          data-testid="li"
        >
          <a
            href="/#introduction"
          >
            About
          </a>
        </li>
        <li
          className="nav__elt"
          data-testid="li"
        >
          <a
            href="/blog"
          >
            Blog
          </a>
        </li>
      </ul>
    </nav>
    <h1
      className="Layout__title"
      data-testid="title"
    />
    <div
      className="Header_separator-white"
      data-testid="separator"
    />
  </div>
  <main />
</div>
`;
